src/keygen/keyshare.rs

```diff
@@ -51,9 +60,34 @@ impl KeySharePublic {
         }
     }
 
+    /// Get the public curvepoint which is the public key share of the
+    /// participant.
+    pub fn X(&self) -> CurvePoint {
+        self.X
+    }
+
     /// Get the ID of the participant who claims to hold the private share
     /// corresponding to this public key share.
     pub fn participant(&self) -> ParticipantIdentifier {
         self.participant
     }
+
+    /// Generate a new [`KeySharePrivate`] and [`KeySharePublic`].
+    pub fn new_keyshare<R: RngCore + CryptoRng>(
+        participant: ParticipantIdentifier,
+        rng: &mut R,
+    ) -> Result<(KeySharePrivate, KeySharePublic)> {
+        let order = k256_order();
+        let random_bn = BigNumber::from_rng(&order, rng);
+        let private_share = KeySharePrivate::set_x(random_bn);
+        /*let private_share = KeySharePrivate {
+            x: BigNumber::from_rng(&order, rng),
+        };*/
+        let public_share = private_share.public_share()?;
+
+        Ok((
+            private_share,
+            KeySharePublic::new(participant, public_share),
+        ))
+    }
```

Comment:
suggestion: if you decide to change the `public_share` function to return a `KeySharePublic` instead of a `CurvePoint`, I think we should take out this method and replace it with direct calls to the constructors, like
```
let private_share = KeySharePrivate::random(&mut rng);
let public_share = private_share.public_share();
```

That is sort of the culminating success of this ticket -- being able to use the keyshare interfaces directly without needing extra functions to get the types right.